epoch.go

```diff
@@ -2737,42 +2890,35 @@ func (e *Epoch) handleReplicationResponse(resp *ReplicationResponse, from NodeID
 		return nil
 	}
 
-	e.Logger.Debug("Received replication response", zap.Stringer("from", from), zap.Int("num seqs", len(resp.Data)), zap.Stringer("latest round", resp.LatestRound))
+	e.Logger.Debug("Received replication response", zap.Stringer("from", from), zap.Int("num seqs", len(resp.Data)), zap.Stringer("latest round", resp.LatestRound), zap.Stringer("latest seq", resp.LatestSeq))
 	nextSeqToCommit := e.nextSeqToCommit()
 
-	validRounds := make([]QuorumRound, 0, len(resp.Data))
 	for _, data := range resp.Data {
-		if err := data.IsWellFormed(); err != nil {
-			e.Logger.Debug("Malformed Quorum Round Received", zap.Error(err))
-			continue
-		}
-
-		if data.EmptyNotarization == nil && nextSeqToCommit > data.GetSequence() {
-			e.Logger.Debug("Received quorum round for a seq that is too far behind", zap.Uint64("seq", data.GetSequence()))
-			continue
-		}
-
-		if data.GetSequence() > nextSeqToCommit+e.maxRoundWindow {
+		if data.Finalization != nil && data.GetSequence() > nextSeqToCommit+e.maxRoundWindow {
 			e.Logger.Debug("Received quorum round for a seq that is too far ahead", zap.Uint64("seq", data.GetSequence()))
 			// we are too far behind, we should ignore this message
 			continue
 		}
 
-		if err := e.verifyQuorumRound(data, from); err != nil {
-			e.Logger.Debug("Received invalid quorum round", zap.Uint64("seq", data.GetSequence()), zap.Stringer("from", from))
+		// We may be really far behind, so we shouldn't process sequences unless they are the nextSequenceToCommit
+		if data.GetRound() > e.round+e.maxRoundWindow && data.GetSequence() != nextSeqToCommit {
```

Comment:
I don't understand why would `data.GetSequence()` be necessary equal to the next sequence to commit?